### ISSUE TITLE
[Bug]: Query with count(*) and querynode panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#23241]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230404-940ead20
- Deployment mode(standalone or cluster): cluster 
- MQ type(rocksmq, pulsar or kafka):  pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

test query count(*) and querynode panic:
```
    def test_count_bounded_level(self):
        """
        target: test count(*) with bounded level
        method: 1. create collection with bounded level
                2. load collection
                3. insert and count
                4. wait bounded time and count
        expected: expected count
        """
        collection_w = self.init_collection_wrap(name=cf.gen_unique_str(prefix), consistency_level=CONSISTENCY_BOUNDED)
        # load collection
        collection_w.create_index(ct.default_float_vec_field_name, index_params=ct.default_flat_index)
        collection_w.load()
        tmp_nb = 100
        df = cf.gen_default_dataframe_data(tmp_nb)
        collection_w.insert(df)
        time.sleep(5)

        count, _ = collection_w.query(expr=default_expr, output_fields=[default_count_output])
        # collection_w.query(expr=default_term_expr, output_fields=[default_count_output])
        # log.info(count)
```
panic logs:  

[qn_pre.log](https://github.com/milvus-io/milvus/files/11167120/qn_pre.log)

```
[2023/04/06 06:31:24.020 +00:00] [DEBUG] [segments/validate.go:52] ["read target partitions"] [traceID=1002b3bab2807cd5219a7970af49b355] [collectionID=440598985650139480] [partitionIDs="[440598985650139481]"]
[2023/04/06 06:31:24.021 +00:00] [DEBUG] [segments/segment.go:401] ["do retrieve on segment"] [collectionID=440598985650139480] [partitionID=440598985650139481] [segmentID=440598985650139514] [msgID=440601853032136705] [segmentType=Growing]
[2023/04/06 06:31:24.021 +00:00] [DEBUG] [segments/segment.go:401] ["do retrieve on segment"] [collectionID=440598985650139480] [partitionID=440598985650139481] [segmentID=440598985650139513] [msgID=440601853032136705] [segmentType=Growing]
[2023/04/06 06:31:24.021 +00:00] [DEBUG] [segments/segment.go:414] ["retrieve result"] [collectionID=440598985650139480] [partitionID=440598985650139481] [segmentID=440598985650139514] [resultNum=0]
[2023/04/06 06:31:24.021 +00:00] [DEBUG] [segments/segment.go:414] ["retrieve result"] [collectionID=440598985650139480] [partitionID=440598985650139481] [segmentID=440598985650139513] [resultNum=0]
[2023/04/06 06:31:24.021 +00:00] [DEBUG] [segments/result.go:279] [mergeSegcoreRetrieveResults] [traceID=1002b3bab2807cd5219a7970af49b355] [limit=-1] [resultNum=1]
[2023/04/06 06:31:24.021 +00:00] [DEBUG] [segments/result.go:279] [mergeSegcoreRetrieveResults] [traceID=1002b3bab2807cd5219a7970af49b355] [limit=-1] [resultNum=1]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19c9120]

goroutine 45649 [running]:
panic({0x3616a60, 0x52b91a0})
    /usr/local/go/src/runtime/panic.go:941 +0x397 fp=0xc002cb5418 sp=0xc002cb5358 pc=0x12b9cf7
runtime.panicmem(...)
    /usr/local/go/src/runtime/panic.go:220
runtime.sigpanic()
    /usr/local/go/src/runtime/signal_unix.go:818 +0x336 fp=0xc002cb5468 sp=0xc002cb5418 pc=0x12d0d16
github.com/milvus-io/milvus/internal/util/typeutil.GetSizeOfIDs(0x3e21390?)
    /go/src/github.com/milvus-io/milvus/internal/util/typeutil/schema.go:652 fp=0xc002cb5470 sp=0xc002cb5468 pc=0x19c9120
github.com/milvus-io/milvus/internal/querynodev2/segments.MergeSegcoreRetrieveResults({0x3e21390?, 0xc002c9bc80?}, {0xc002c5cdb8, 0x1, 0x1?}, 0xffffffffffffffff)
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/result.go:294 +0x2cd fp=0xc002cb56e0 sp=0xc002cb5470 pc=0x2d61bed
github.com/milvus-io/milvus/internal/querynodev2/segments.MergeSegcoreRetrieveResultsAndFillIfEmpty({0x3e21390?, 0xc002c9bc80?}, {0xc002c5cdb8?, 0xc000931e60?, 0x0?}, 0x0?, {0x0, 0x0, 0x0}, 0xc0028ccb40)
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/result.go:367 +0x45 fp=0xc002cb5738 sp=0xc002cb56e0 pc=0x2d622c5
github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).querySegments(0xc000f4ad80, {0x3e21390, 0xc002c9bc80}, 0xc002cabf80)
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/handlers.go:240 +0x2bd fp=0xc002cb5810 sp=0xc002cb5738 pc=0x2da7a1d
github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).queryChannel(0xc000f4ad80, {0x3e21390, 0xc002c9bb40}, 0xc002cabf80, {0xc002627fc0, 0x33})
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/handlers.go:140 +0x966 fp=0xc002cb5f10 sp=0xc002cb5810 pc=0x2da63a6
github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).Query.func1()
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/services.go:714 +0x50 fp=0xc002cb5f78 sp=0xc002cb5f10 pc=0x2db65f0
golang.org/x/sync/errgroup.(*Group).Go.func1()
    /go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64 fp=0xc002cb5fe0 sp=0xc002cb5f78 pc=0x2699d24
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc002cb5fe8 sp=0xc002cb5fe0 pc=0x12ef0c1
created by golang.org/x/sync/errgroup.(*Group).Go
    /go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5

goroutine 1 [chan receive, 98 minutes]:
runtime.gopark(0xc00110f728?, 0xc00110f780?, 0xd3?, 0xb3?, 0xc00110f780?)
    /usr/local/go/src/runtime/proc.go:361 +0xd6 fp=0xc0014f3750 sp=0xc0014f3730 pc=0x12bcf56
runtime.chanrecv(0xc00077e000, 0xc00110fbe8, 0x1)
    /usr/local/go/src/runtime/chan.go:577 +0x56c fp=0xc0014f37e0 sp=0xc0014f3750 pc=0x1288d2c
runtime.chanrecv1(0xc00077e000?, 0xc00110fdb8?)
    /usr/local/go/src/runtime/chan.go:440 +0x18 fp=0xc0014f3808 sp=0xc0014f37e0 pc=0x1288758
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc00110fe58, 0x0, {0xc000dd5260?, 0xd?})
    /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:325 +0xa6c fp=0xc0014f3e08 sp=0xc0014f3808 pc=0x2f29d4c
github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc000598180, {0xc000216060?, 0x3, 0x3}, 0xc000590fc0)
    /go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:112 +0x66e fp=0xc0014f3ee0 sp=0xc0014f3e08 pc=0x2f363ce
github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc000216060?, 0x3, 0x3})
    /go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e fp=0xc0014f3f58 sp=0xc0014f3ee0 pc=0x2f35c5e
main.main()
    /go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e fp=0xc0014f3f80 sp=0xc0014f3f58 pc=0x2f38aae
runtime.main()
    /usr/local/go/src/runtime/proc.go:250 +0x212 fp=0xc0014f3fe0 sp=0xc0014f3f80 pc=0x12bcb92
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc0014f3fe8 sp=0xc0014f3fe0 pc=0x12ef0c1
```

### Milvus Log

devops cluster, chaos-testing ns:
```
cluster-count-milvus-datacoord-764bc747d5-2xd69                  1/1     Running       0                3h6m
cluster-count-milvus-datanode-674cffb9d7-cpfc9                   1/1     Running       0                3h6m
cluster-count-milvus-indexcoord-85848df967-47lvd                 1/1     Running       0                3h6m
cluster-count-milvus-indexnode-77587cf94f-wrzst                  1/1     Running       0                3h6m
cluster-count-milvus-proxy-5f88b547bd-8vgz6                      1/1     Running       0                3h6m
cluster-count-milvus-querycoord-587876f9dc-cpkgf                 1/1     Running       0                3h6m
cluster-count-milvus-querynode-5d9c44fd66-ljkzk                  1/1     Running       3 (3m53s ago)    3h6m
cluster-count-milvus-rootcoord-6fbdcb89bf-r6r92                  1/1     Running       0                3h6m
```

**Comment by longjiquan** [comment]
Will be fixed by #23268

**Comment by ThreadDao** [comment]
fixed
